group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupConfig.java

```diff
@@ -57,13 +62,20 @@ public class GroupConfig extends AbstractConfig {
             GroupCoordinatorConfig.CONSUMER_GROUP_HEARTBEAT_INTERVAL_MS_DEFAULT,
             atLeast(1),
             MEDIUM,
-            GroupCoordinatorConfig.CONSUMER_GROUP_HEARTBEAT_INTERVAL_MS_DOC);
+            GroupCoordinatorConfig.CONSUMER_GROUP_HEARTBEAT_INTERVAL_MS_DOC)
+        .define(SHARE_RECORD_LOCK_DURATION_MS_CONFIG,
+            INT,
+            ShareGroupConfig.SHARE_GROUP_RECORD_LOCK_DURATION_MS_DEFAULT,
+            atLeast(1000),
```

Comment:
Apologies, but in `ShareGroupConfig`, the validator for `SHARE_GROUP_RECORD_LOCK_DURATION_MS_CONFIG` is set to `between(1000, 60000)`, while in `GroupConfig` there’s no upper bound. Why is that?

Additionally, the validator `between(1000, 60000)` seems odd, given that the upper bound for `SHARE_GROUP_MAX_RECORD_LOCK_DURATION_MS_CONFIG` is `3600000`. Shouldn't the correct validator be `between(1000, 3600000)`?

Reply:
There is a hierarchy of configs to bear in mind, as described in KIP-932.

The broker config `group.share.record.lock.duration.ms` is the default duration when there is no group-specific configuration. This one should be `between(1000, 60000)`. Most groups will use this config.

The broker configs `group.share.min.record.lock.duration.ms` and `group.share.max.record.lock.duration.ms` control the broker's min and max which are used to bound group-specific configuration of the lock duration. Static validation of `atLeast(1000)` is certainly not wrong but the majority of the validation is dynamic in the `validateValues(,,)` method.

Reply:
> The broker config group.share.record.lock.duration.ms is the default duration when there is no group-specific configuration. This one should be between(1000, 60000). Most groups will use this config.

My point was why `group.share.record.lock.duration.ms` has a upper bound of `60000` rather `3600000`? For example, the following configs are illegal.
```
group.share.min.record.lock.duration.ms=30000
group.share.record.lock.duration.ms=65000
group.share.max.record.lock.duration.ms=70000
```

I assumed the rules is 
```
group.share.min.record.lock.duration.ms <= group.share.record.lock.duration.ms <= group.share.max.record.lock.duration.ms
```
However, the actual rule is 
```
group.share.min.record.lock.duration.ms <= group.share.record.lock.duration.ms <= min(60000, group.share.max.record.lock.duration.ms)
```

Reply:
I think the idea is to have a default config `group.share.min.record.lock.duration.ms` between 1 and 60 secs.

But this can only be changed dynamically using min and max setting, and can be between 1 sec to 1 hour. But then the lock duration can only be set dynamically.

Reading it again, I also felt it's a bit confusing and it might be better to set default config (`group.share.record.lock.duration.ms`)  also from 1 sec to 1 hour. Would be good to have @AndrewJSchofield thoughts as he might have already thought about it and I could be missing something.

Reply:
Thanks for the comments @chia7712 and @apoorvmittal10. This is working as I intended, but I can see that people are finding it confusing. I think it sensible to adjust the maximum for `group.share.record.lock.duration.ms` to 3600000 (1 hour) as suggested. I'll change the KIP and put in a PR.